src/handshake/client.rs

```diff
@@ -90,31 +103,45 @@ impl<S: Read + Write> HandshakeRole for ClientHandshake<S> {
                 result,
                 tail,
             } => {
-                self.verify_data.verify_response(&result)?;
+                let mut config = self.config.take().unwrap();
```

Comment:
Oh, that looks like a bold move :) Are we sure that it's always `Some`? Because if we do, then probably `config` must not be an `Option`.